planner/core/find_best_task.go

```diff
@@ -1227,9 +1227,15 @@ func (ds *DataSource) crossEstimateRowCount(path *util.AccessPath, expectedCnt f
 	if err != nil {
```

Comment:
For your case, the `isFull` in line 1218 is expected to be true, so this function is supposed to return at line 1219?

Reply:
no, it return false.

Reply:
Why? it is not as expected, we should fix the problem there?

Reply:
https://github.com/fzhedu/tidb/blob/5e0e508b4ba605231f1e6e17389dcbfef4ce7486/planner/core/find_best_task.go#L1165
ranges : [0, +oo)
rangeCounts[0] > 1
It seems a bug: https://github.com/fzhedu/tidb/blob/5e0e508b4ba605231f1e6e17389dcbfef4ce7486/planner/core/find_best_task.go#L1169 should before the `break` of #L1167